ql/instruments/bonds/fixedratebond.hpp

```diff
@@ -95,10 +97,11 @@ namespace QuantLib {
                       BusinessDayConvention exCouponConvention = Unadjusted,
                       bool exCouponEndOfMonth = false);
         Frequency frequency() const { return frequency_; }
-        const DayCounter& dayCounter() const { return dayCounter_; }
+        const DayCounter& dayCounter(const Date& d=Date());
```

Comment:
I would probably leave the method as it is, and add another inspector `firstPeriodDayCounter`.

Reply:
Will do